pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -288,8 +298,9 @@ func NewKV(cfg KVConfig) (*KV, error) {
 			return nil, err
 		}
 
-		if err != nil {
-			level.Error(util.Logger).Log("msg", "failed to join memberlist cluster", "err", err)
+		if reached == 0 && cfg.MaxJoinRetries > 0 {
+			level.Debug(util.Logger).Log("msg", "failed to join memberlist cluster, keep trying in the background", "node", cfg.NodeName, "err", err)
+			go mlkv.retryJoinWithBackoff(cfg.JoinMembers)
```

Comment:
Before this PR there was no retry and, in case of failure joining the cluster, the creation of the ring client was failing causing a cascading failure. This PR adds background retries, but in case it fails to join the cluster after all retries, the error is just logged so the system will keep running but not functioning correctly. Am I missing anything?

Reply:
You're not missing anything. Me and Periklis agreed on this design until we have memberlist in service mode ready.

For now, we don't want to block `NewKV` method, so we only check the abort flag on first failure.

Once memberlist KV is converted to a service, it will do joining with backoff in `Starting` state, and if that fails and Abort flag is set, it will report error.

Reply:
Ok. Sorry for having missed the conversation (I may have checked). Then LGTM.

Reply:
It was private conversation, sorry for not including it here.